### DISCUSSION TITLE
How to share atom between Provider and custom hook

### DISCUSSION THREAD

**Discussion by DonaldTaylor** [pmndrs/jotai#1327] [General]
I have a library that wraps itself in a Provider: `<Provider> /* my library code */ </Provider>`.

But I have a custom hook, e.g., `useMyLibraryHook`, associated with this library that needs to access the atoms under `<Provider />`.

If I don't use a `<Provider />`, the atoms are "global" and are accessible. But under a `<Provider />`, I can't find out how to access them.

How can I access the atoms within the custom hook?

Thank you very much!

**Comment by dai-shi** [comment]
Provide `scope` is for you.

```jsx
<Provider scope="foo">
  ...
</Provider>

useAtom(anAtom, "foo") // will use the closest Provider with "foo" scope in the component tree.
```

**Comment by DonaldTaylor** [comment]
Thank you!
